Mathlib/Tactic/ToFun.lean

```diff
@@ -0,0 +1,47 @@
+/-
+Copyright (c) 2025 Jovan Gerbscheid. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jovan Gerbscheid
+-/
+module
+
+public meta import Mathlib.Util.AddRelatedDecl
+public meta import Mathlib.Tactic.Push
+
+/-!
+# The `to_fun` attribute
+
+Adding `@[to_fun]` to a lemma named `F` creates a new lemma named `fun_F`, which is obtained by
+running `pull fun _ ↦ _` on the type of `F`.
+-/
+
+public meta section
+
+open Lean Meta Elab Tactic
+namespace Mathlib.Tactic
+
+/--
+Adding `@[to_fun]` to a lemma
+```
+theorem Continuous.mul (hf : Continuous f) (hg : Continuous g) : Continuous (f * g)
+```
+will generate a new lemma `Continuous.fun_mul` with conclusion `Continuous fun x => f x * g x`.
+
+Use the `to_fun (attr := ...)` syntax to add the same attribute to both declarations.
+-/
+syntax (name := to_fun) "to_fun" (" (" &"attr" " := " Parser.Term.attrInstance,* ")")? : attr
+
+initialize registerBuiltinAttribute {
+  name := `to_fun
+  descr := ""
+  applicationTime := .afterCompilation
+  add := fun src ref kind => match ref with
+  | `(attr| to_fun $[(attr := $stx?,*)]?) => MetaM.run' do
+    if (kind != AttributeKind.global) then
+      throwError "`to_fun` can only be used as a global attribute"
```

Comment:
Can you add a test covering this, please?

Reply:
Would you like to add a test for reassoc and to_additive? :-) If I see correctly, that should be easy to do.